ocaml/xcp-rrdd/scripts/rrdd/rrdd.py

```diff
@@ -310,20 +311,27 @@ def update(self):
         metadata_json = json.dumps(metadata, sort_keys=True).encode('utf-8')
         metadata_checksum = crc32(metadata_json) & 0xffffffff
 
-        self.dest.seek(0)
-        self.dest.write('DATASOURCES'.encode())
-        self.dest.write(pack(">LLLQ",
-                             data_checksum,
-                             metadata_checksum,
-                             len(self.datasources),
-                             timestamp))
+        # First write the updated data and metadata
```

Comment:
so what happens if the new data & metadata is shorter than the previous one? I imagine it will leave some garbage after the end of the metadata (left over from previous data)?

Reply:
which won't be read as the header specifies how long the metadata is in the `metadata length` field. And the metadata should only be re-read if the `timestamp` field of the header changes.

Reply:
Yes, but that's based on the assumption that the header is being written as well as the data. If the data is written, but the header has not yet been written, i.e. the header of the previous data would be read and incorrect amount of data will be read in that case.